prov/shm/src/smr_msg.c

```diff
@@ -131,10 +198,14 @@ static ssize_t smr_recvmsg(struct fid_ep *ep_fid, const struct fi_msg *msg,
 
 	ep = container_of(ep_fid, struct smr_ep, util_ep.ep_fid.fid);
 
+	flags |= ep->util_ep.rx_msg_flags;
+	if (flags & FI_MULTI_RECV)
+		return smr_generic_mrecv(smr_get_smr_srx(ep),
+					 msg->msg_iov, msg->desc, msg->iov_count, 
+					 msg->addr, msg->context, flags);
```

Comment:
Why not move the FI_MULTI_RECV check info smr_generic_recv, rather than duplicating it before each call to it?

Reply:
I had it like that before but moved it to remove the check from the tagged path which never uses multi recv and previously an app could call tagged + multi recv and it would pass but have issues later. I can just add an assert though instead.